hyperspy/io_plugins/blockfile.py

```diff
@@ -287,7 +343,12 @@ def file_writer(filename, signal, **kwds):
                                         ('ID', endianess + 'u4')])
         dp_head['MAGIC'] = 0x55AA
         # Write by loop:
-        for img in signal._iterate_signal():
+        for img in tqdm(signal._iterate_signal(),
```

Comment:
Since I got rid of the loop, I'm using something modeled after https://github.com/hyperspy/hyperspy/blob/2ddc359c5a6c0ff300507c383dfcc1de644ff11a/hyperspy/_signals/lazy.py#L121 with the dask progress bar. There is no more progress bar possible with non-lazy datasets.